File: examples/reticulated_python/15.py
class A:
	a = 1

class B(A):
	def foo(self:B) -> Int:
		return 2

class C(A):
	def bar(self:C) -> Int:
		return 3

@fields({'a' : A, 'b' : B, 'c' : C})
class C:
	a = A()
	b = B()
	c = C()
	def main(self:C) -> Int:
		self.b = B()
		self.a = self.b
		self.c = self.a
		return self.b.foo()


print(C().main())

File: examples/strongscript/15.ts
class A {
}

class B extends A {
	foo():!number { return 2 }

}
class C extends A {
	bar():!number { return 3 }
}

class D {
	a:!A;
	b:!B;
	c:!C;
	main():!number {
		this.b = <!B>new B()
		this.a = this.b
		this.c = <!C>this.a
		return this.b.foo()
	}
}

new D().main()